### DISCUSSION TITLE
docs/other/vpn

### DISCUSSION THREAD

**Discussion by giscus** [AMD-NICK/blog.amd-nick.me#16] [Docs comments]
# docs/other/vpn

Tailscale demo cover

https://blog.amd-nick.me/docs/other/vpn

**Comment by SVG62** [comment]
Лучшее решение, Wireguard+docker

**Reply by AMD-NICK** [reply]
Tailscale работает поверх Wireguard, имея при себе несколько дополнительных наворотов. Из тех, что я пользуюсь:

- Подключение к SSH по `ssh user@название_хоста_в_tailscale_панели`. Например, `ssh root@blabla`. Вместо ввода пароля или установки SSH ключа на хост, нужно просто нажать на ссылку в консоли для авторизации. Делать это нужно не постоянно, а только если долго не авторизировался.
- Удобная передача файлов между всеми устройствами с установленным Tailscale: телефоны, планшеты, ноутбуки, домашние ПК, сервера. На компе `ПКМ > Отправить > Tailscale > Нужный хост`, на нужном хосте (если linux): `tailscale file get .`. А если передача на винду или iphone или мак, то вылезет удобное окошко
- Любое из устройств может быть exit node (то, что большинство понимает как VPN, хотя VPN это не только когда у тебя IP меняется). Можно даже быть в Франции, а выходить в интернет с IP домашнего компа, который стоит в условной Германии
- Этим не пользуюсь, но как я понимаю, на этом он сделал основной акцент: ACL (настройки прав доступа). Можно разным пользователям или группам выдать разные гибкие права доступа к разным узлам Tailscale сети. Даже к разным портам. Все это через удобную панель с Live обновлениями на сайте Tailscale
- Tailscale client/server кроссплатформенный и используется и без Docker. Как я уже говорил, можно хоть iPhone сделать частью VPN сети

**Reply by SVG62** [reply]
Просто докер удобен в установке а количество выдаваемых конфигов можно менять на лету

**Reply by SVG62** [reply]
На сколько я помню при тесте Tailscale у меня на теле переставали работать банковские приложения и отвалы были, и жор батареи, а вот с wireguard пашет как часы. Я давно все доступы по RDP,VNC,SSH перевел в докер через guacamole и закрыл все порты за ненадобностью проброса портов в целях безопсности.

**Reply by AMD-NICK** [reply]
Сделал сборник всякого по Docker. Возможно, захочешь дополнить:
https://blog.amd-nick.me/docs/other/docker

**Comment by shockpast** [comment]
# GoodbyeDPI
Бесплатный сервис, почти заменяющий другие ВПН-ы.

## Отзыв
Пользуюсь GoodbyeDPI по причине того что не хочется покупать ВПН-ы и подобные сервисы, так-как в моих повседневных целях он абсолютно нужен.
GoodbyeDPI - отлично справляется с работой по обходу блокировок, которые были специально сделаны вашим провайдером, говоря об его аптайме или перебоях в работе интернета, то такого ни разу не случалось.

Довольно скучный и бесполезный факт, но его можно запускать в двух вариантах:
- Сервис (тихий режим, скрытый и не мешает работе)
- Консольное приложение (показывает дополнительную информацию)

В принципе, мои знания по составлению сочинений были всегда очень скудны, многого рассказать не смогу, так что лучше заглядывайте на репозиторий "гудбайдпи" и прочитайте "ридми", там более понятнее и яснее рассказано.

## Отличия
- IP-Адрес остаётся не изменённым, так-как обход блокировок сайтов, является немного другой.
- Бесплатный.
- Опен-Сурс.
